### PR TITLE
Add support for Cisco Aironet WLC

### PR DESCRIPTION
**Pull request by mpenning**
Please consider this pull request for inclusion in exscript.  Most changes were in src/Exscript/protocols/drivers/aironet.py

Test results follow...

    (py27_exscript)[mpenning@tsunami exscript]$ cd tests/Exscript
    (py27_exscript)[mpenning@tsunami Exscript]$ python protocols/OsGuesserTest.py
    testConstructor (__main__.OsGuesserTest) ... ok
    testDataReceived (__main__.OsGuesserTest) ... ok
    testGet (__main__.OsGuesserTest) ... ok
    testReset (__main__.OsGuesserTest) ... ok
    testSet (__main__.OsGuesserTest) ... ok
    testSetFromMatch (__main__.OsGuesserTest) ... ok

    ----------------------------------------------------------------------
    Ran 6 tests in 1.128s

    OK
    (py27_exscript)[mpenning@tsunami Exscript]$

### CODE REVIEW

**Comment by egroeper** [comment]
make tests doesn't finish successfully:
```
Ran 500 tests in 931.033s

FAILED (failures=2, errors=40)
```

At the moment I'm not sure, what the problem is. I think it could be your banner files interfering with generic driver. But OsGuesser tests run fine...

Most errors are:
```
[...]
    raise TimeoutException(msg)
TimeoutException: Buffer: 'sab\r'
```
I think there are problems detecting the prompts.

File: src/Exscript/protocols/drivers/aironet.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        self.prompt_re   = _prompt_re
+        #self.error_re    = _error_re
+
+    def check_head_for_os(self, string):
+        if '(Cisco Controller)' in string:
+            return 60
+        if _tacacs_re.search(string):
+            return 50
+        if _user_re[0].search(string):
+            return 30
```
Comment (egroeper) [inline src/Exscript/protocols/drivers/aironet.py]: OSGuesserTest.py doesn't find it, but there is a problem with your os detection.
```_tacacs_re```  is equal to ios driver. That's a problem, if detection is based on that.
Your ```_user_re``` is the reason why the tests fail.

I suggest you remove the ```_tacacs_re``` and ```_user_re``` checks.
If the "(Cisco Controller)" check doesn't work, you could set the driver manually.

**Comment by egroeper** [comment]
Thanks for your contribution. If you fix the os detection, I would be happy to merge your pull request.

**Comment by mpenning** [comment]
I attempted to sync everything to the master and update with your suggested changes, but other tests fail now... I'm not sure this was my problem, but this is what I see when I `make tests`...

    (py27_test)[mpenning@tsunami exscript]$ make tests
    cd tests/Exscript/; ./run_suite.py 1
    .................................../templates/stdlib.string/pseudodev.py error: Buffer: 'sab\r\n'
    Traceback (most recent call last):
      File "/home/mpenning/blah/exscript/tests/Exscript/../../src/Exscript/workqueue/Job.py", line 64, in run
        self.function(self)
      File "/home/mpenning/blah/exscript/tests/Exscript/../../src/Exscript/Queue.py", line 80, in _wrapped
        result = func(job, host, conn, *args, **kwargs)
      File "/home/mpenning/blah/exscript/tests/Exscript/../../src/Exscript/util/decorator.py", line 37, in decorated
        return function(*(inner_args + args), **kwargs)
      File "/home/mpenning/blah/exscript/tests/Exscript/TemplateTest.py", line 37, in dummy_cb
        conn.login(flush = True)
      File "/home/mpenning/blah/exscript/tests/Exscript/../../src/Exscript/protocols/Protocol.py", line 606, in login
        self.authenticate(account, flush = False)
      File "/home/mpenning/blah/exscript/tests/Exscript/../../src/Exscript/protocols/Protocol.py", line 630, in authenticate
        self.app_authenticate(app_account, flush = flush)
      File "/home/mpenning/blah/exscript/tests/Exscript/../../src/Exscript/protocols/Protocol.py", line 798, in app_authenticate
        self._app_authenticate(account, password, flush, bailout)
      File "/home/mpenning/blah/exscript/tests/Exscript/../../src/Exscript/protocols/Protocol.py", line 702, in _app_authenticate
        raise TimeoutException(msg)
    TimeoutException: Buffer: 'sab\r\n'

**Comment by mpenning** [comment]
I am new to exscript's OS detection requirements... When I `cd tests/` and `make tests`, I get a clean run now.  Thank you for your patience.

    error: Socket is closed
    ...............................................................................................................................................................    .............................................................................................................................
    ----------------------------------------------------------------------

    Ran 500 tests in 199.315s

    OK

**Comment by egroeper** [comment]
Thanks for your contribution. 

For the future: Consider using feature branches for your contributions.
That makes contributing and merging upstream easier.

**Pull request merged**